mail_tracking/models/mail_tracking_email.py

```diff
@@ -216,13 +218,25 @@ def _partners_email_bounced_set(self, reason, event=None):
 
     @api.multi
     def smtp_error(self, mail_server, smtp_server, exception):
-        self.sudo().write({
-            'error_smtp_server': tools.ustr(smtp_server),
-            'error_type': exception.__class__.__name__,
-            'error_description': tools.ustr(exception),
-            'state': 'error',
-        })
-        self.sudo()._partners_email_bounced_set('error')
+        values = {}
+        IrMailServer = self.env['ir.mail_server']
+        if str(exception) == IrMailServer.NO_VALID_RECIPIENT \
+                and not self.recipient_address:
+            values.update({
+                'state': 'error',
+                'error_type': 'no_recipient',
+                'error_description':
+                    _("The partner doesn't have a defined email"),
```

Comment:
I think you don't want translations in a log:

```suggestion
                    "The partner doesn't have a defined email",
```

Otherwise your chinese colleague could produce logs impossible to understand to you, or viceversa.